pkg/operations/armhelpers/azureclients.go

```diff
@@ -0,0 +1,32 @@
+package armhelpers
+
+import (
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// AzureClients contains Azure clients
+type AzureClients struct {
+	SubscriptionID   string
+	AzureEnvironment azure.Environment
+	TenantID         string
+
+	GroupsClient *resources.GroupsClient
+	VMClient     *compute.VirtualMachinesClient
+}
+
+// Create method creates various Azure clients
+func (ac *AzureClients) Create(token *adal.ServicePrincipalToken) (*AzureClients, error) {
```

Comment:
Yes, I agree and the RP already does this (https://msazure.visualstudio.com/DefaultCollection/One/_git/Compute-containerService?path=%2Fsrc%2Fmicrosoft.com%2Fcontainerservice%2Ftemplates%2Fhelper.go&version=GBmaster&_a=contents). 

I tried writing something similar but reverted back to this because I wanted to see it working :). Do you want me to do same thing in this PR?